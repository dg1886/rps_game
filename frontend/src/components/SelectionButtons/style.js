import { Typography } from "@mui/material";
import styled, { keyframes } from "styled-components";

import img from "../../assets/images/bg-triangle.svg";

const animate = keyframes`
  0%, 18%, 20%, 50.1%, 65.1%, 80%, 90.1%, 92% {
    color: ${({ theme }) => theme.palette.text.secondary};
    text-shadow: none;
  }
  18.1%, 20.1%, 30%, 50%, 60.1%, 65%, 80.1%, 90%, 92.1%, 100% {
    color: #fff;
    text-shadow: 0 0 10px #03bcf4,
    0 0 20px #03bcf4,
    0 0 40px #03bcf4,
    0 0 80px #03bcf4,
    0 0 160px #03bcf4;
  }
`;

export const IconGridWrap = styled.div`
  display: grid; 
  margin-top: 5%;
  grid-template-columns: 1fr 1fr 1fr; 
  grid-template-rows: 1fr 1fr 1fr; 
  gap: 0px 0px; 
  grid-template-areas: 
      "leftUp . rightUp"
      ". . ."
      ". down ."; 
  background-image: url(${img});
  background-repeat: no-repeat;
  background-size: 80%;
  background-position-x: 50%;
  background-position-y: 50%;
  @media (max-width: 450px) {
    margin-top: 30%;
  }
`;

export const AnimatedTittle = styled(Typography)`
  position: relative;
  margin-top: 25px;
  color: transparent;
  text-transform: uppercase;
  width: 100%;
  text-align: center;
  line-height: 2rem;
  font-size: 2.5rem;
  animation: ${animate} 6s linear;
  animation-iteration-count: infinite;

  ${({ theme }) => theme.breakpoints.down("sm")} {
    font-size: 2rem;
    line-height: 1.5rem;
  }
`;
