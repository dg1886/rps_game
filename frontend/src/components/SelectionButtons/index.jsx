import { Grid } from "@mui/material";
import { useContext, useEffect } from "react";
import { io } from "socket.io-client";

import { choiceIcons } from "../../constants/choiceIcons";
import { GAME_ITEMS } from "../../constants/names";
import { GameContext } from "../../services/gameContext";
import IconButton from "../Button";
import { AnimatedTittle, IconGridWrap } from "./style";

const iconsInfo = [
  { area: "leftUp", id: GAME_ITEMS.ROCK },
  { area: "rightUp", id: GAME_ITEMS.PAPER },
  { area: "down", id: GAME_ITEMS.SCISSORS },
];

const SelectionButtons = () => {
  const { setMyChoice, toggleBattle, myChoice } = useContext(GameContext);

  useEffect(() => {
    console.log(myChoice);
    const socket = io("/");
    socket.on("connect", () => {
      socket.emit("battle", myChoice);
    });
  }, [myChoice]);

  const iconEvent = (e) => { toggleBattle(true); setMyChoice(e.currentTarget.id); };

  return (
    <Grid
      container
      xs={12}
      sx={{
        flexDirection: "column", flexWrap: "nowrap", justifyContent: "flex-start", alignItems: "center", height: "100%",
      }}
    >
      <AnimatedTittle spacing={45} variant="caption">make your choice</AnimatedTittle>

      <IconGridWrap>
        {iconsInfo.map((item) => (
          <IconButton id={item.id} variant="iconWrap" figure={item.id} onClick={iconEvent} gridArea={item.area} isShake>
            {choiceIcons[item.id]}
          </IconButton>
        ))}
      </IconGridWrap>

    </Grid>
  );
};

export default SelectionButtons;
