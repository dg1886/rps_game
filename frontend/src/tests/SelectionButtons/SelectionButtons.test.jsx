import { screen } from "@testing-library/react";

import SelectionButtons from "../../components/SelectionButtons";
import renderWithContext from "../customRenderWithContext";

describe("Selection Button component", () => {
  test("buttons snapshot", () => {
    const viewTriangleButtons = renderWithContext(<SelectionButtons />);
    expect(viewTriangleButtons).toMatchSnapshot();
  });
  test("text in component", () => {
    renderWithContext(<SelectionButtons />);
    expect(screen.getByText(/make your choice/i)).toBeInTheDocument();
  });
});
