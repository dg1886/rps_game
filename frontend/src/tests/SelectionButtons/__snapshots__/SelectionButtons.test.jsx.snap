// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection Button component buttons snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root-iNTOv hUKhIt MuiGrid-root MuiGrid-container"
      >
        <span
          class="MuiTypography-root-bXavBU bmHdqU MuiTypography-root MuiTypography-caption sc-evZas cHSihy"
          spacing="45"
        >
          make your choice
        </span>
        <div
          class="sc-crXcEl dDKibQ"
        >
          <button
            class="MuiButtonBase-root-JobBs loEBuR MuiButton-root MuiButton-iconWrap MuiButton-iconWrapPrimary MuiButton-sizeMedium MuiButton-iconWrapSizeMedium MuiButtonBase-root MuiButton-root-gywsUZ gICcYZ sc-iqcoie Eaqjx"
            figure="ROCK"
            id="ROCK"
            tabindex="0"
            type="button"
          >
            <svg>
              icon-rock.svg
            </svg>
            <span
              class="MuiTouchRipple-root-iePYqk ifewIH MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root-JobBs loEBuR MuiButton-root MuiButton-iconWrap MuiButton-iconWrapPrimary MuiButton-sizeMedium MuiButton-iconWrapSizeMedium MuiButtonBase-root MuiButton-root-gywsUZ eRViBu sc-iqcoie hVGZMO"
            figure="PAPER"
            id="PAPER"
            tabindex="0"
            type="button"
          >
            <svg>
              icon-paper.svg
            </svg>
            <span
              class="MuiTouchRipple-root-iePYqk ifewIH MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root-JobBs loEBuR MuiButton-root MuiButton-iconWrap MuiButton-iconWrapPrimary MuiButton-sizeMedium MuiButton-iconWrapSizeMedium MuiButtonBase-root MuiButton-root-gywsUZ iVdQmY sc-iqcoie eqtQyh"
            figure="SCISSORS"
            id="SCISSORS"
            tabindex="0"
            type="button"
          >
            <svg>
              icon-scissors.svg
            </svg>
            <span
              class="MuiTouchRipple-root-iePYqk ifewIH MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root-iNTOv hUKhIt MuiGrid-root MuiGrid-container"
    >
      <span
        class="MuiTypography-root-bXavBU bmHdqU MuiTypography-root MuiTypography-caption sc-evZas cHSihy"
        spacing="45"
      >
        make your choice
      </span>
      <div
        class="sc-crXcEl dDKibQ"
      >
        <button
          class="MuiButtonBase-root-JobBs loEBuR MuiButton-root MuiButton-iconWrap MuiButton-iconWrapPrimary MuiButton-sizeMedium MuiButton-iconWrapSizeMedium MuiButtonBase-root MuiButton-root-gywsUZ gICcYZ sc-iqcoie Eaqjx"
          figure="ROCK"
          id="ROCK"
          tabindex="0"
          type="button"
        >
          <svg>
            icon-rock.svg
          </svg>
          <span
            class="MuiTouchRipple-root-iePYqk ifewIH MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root-JobBs loEBuR MuiButton-root MuiButton-iconWrap MuiButton-iconWrapPrimary MuiButton-sizeMedium MuiButton-iconWrapSizeMedium MuiButtonBase-root MuiButton-root-gywsUZ eRViBu sc-iqcoie hVGZMO"
          figure="PAPER"
          id="PAPER"
          tabindex="0"
          type="button"
        >
          <svg>
            icon-paper.svg
          </svg>
          <span
            class="MuiTouchRipple-root-iePYqk ifewIH MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root-JobBs loEBuR MuiButton-root MuiButton-iconWrap MuiButton-iconWrapPrimary MuiButton-sizeMedium MuiButton-iconWrapSizeMedium MuiButtonBase-root MuiButton-root-gywsUZ iVdQmY sc-iqcoie eqtQyh"
          figure="SCISSORS"
          id="SCISSORS"
          tabindex="0"
          type="button"
        >
          <svg>
            icon-scissors.svg
          </svg>
          <span
            class="MuiTouchRipple-root-iePYqk ifewIH MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
